1566 shape feature extract

raylim opened the pull request.

raylim committed:
feat: dsa_viz extract joint object tile

raylim committed:
build: update shapely

raylim committed:
feat: extract tile shape features cli

chore: 377, 375, 374, 379, 382 package versions

feat: extract_tile_shape_features: add options for limiting variables in final output, clean up output a bit

feat: add extra features to extract_tile_shape_features

armaank commented:
This looks very good, good call on computing log ratios

A few items to address:
1. You can remove all features extracted for `Detection probability`, these are unneeded and can be dropped from the fx extraction phase
2. You can remove features associated w/ unclassified regions, those can be dropped from the fx extraction phase
3. The whole slide features are redundant in this output format, since they are the same no matter the cell type, so we have double the number of features for whole slide features
4. A lot of the whole slide features we probably don't need, here is a list of features that don't need to be extracted: anything related to bounding boxes (bbox), anything related to centroids, anything related to inertia tensors, whole slide label, anything related to moments, whole slide area in microns (we already have whole slide area measured in pixels, and it kind of makes sense to keep all of our measurements in pixels rather than just converting area to units of microns)


I also think it makes sense to restructure the output format. Even though it'll be a little wide, it's more convenient for downstream use to have all of these features in a single column, so the columns will be structured as `variable_parent_class` . That way for an entire cohort we can concat all of the single-row results into a dataframe suitable for downstream use in a scikit-learn or pytorch model.

Commit:
Added spatial analysis tutorial notebook

This is just Armaan's original notebook, updated for the current
state of luna.  It still uses the CLIs.

Commit:
Rewrote spatial tutorial notebook as Python code, and removed UMAP calcs

1. Added function to pull output file names from metadata.yml files.
2. Cast Path objects to strings in a couple of places.
3. Check for both `aperio.MPP` and `openslide.mpp-x` in the slide
   image properties, and log a warning if neither is found.
4. Added new notebook to the tutorial docs
5. Renamed notebook to spatial_stats.ipynb

Commit:
Allow non-local connections to the docs server

This lets us run a docs server on one of the compute servers and view
the docs site on our laptops.

raylim committed:
tests: update test_extract_shape_features

tomp reviewed (approved):
Approved
